metrics/prometheus_machine.go

```diff
@@ -90,6 +99,42 @@ func NewPrometheusMachineCollector(i infoProvider) *PrometheusMachineCollector {
 					return metricValues{{value: float64(machineInfo.NumSockets)}}
 				},
 			},
+			{
+				name:        "machine_cpu_cache_capacity_bytes",
+				help:        "Cache size in bytes assigned to NUMA node and CPU core.",
+				valueType:   prometheus.GaugeValue,
+				extraLabels: []string{prometheusNodeLabelName, prometheusCoreLabelName, prometheusTypeLabelName, prometheusLevelLabelName},
+				getValues: func(machineInfo *info.MachineInfo) metricValues {
+					return getCaches(machineInfo)
+				},
+			},
+			{
+				name:        "machine_thread_siblings_count",
+				help:        "Number of CPU thread siblings.",
+				valueType:   prometheus.GaugeValue,
+				extraLabels: []string{prometheusNodeLabelName, prometheusCoreLabelName, prometheusThreadLabelName},
```

Comment:
what is the cardinality of each of these?  This may end up being a very expensive metric...

Reply:
number of `machine_thread_siblings_count` metrics will be equal to number of CPU threads on the machine, on my machine 72...
Adding `machine_thread_siblings_count` metric I thought about exposing the most information from topology and at this moment I don't have better idea for it so I think we can get rid of this metric for now.

Reply:
I see that similar case will be for `machine_cpu_cache_capacity_bytes`, on my machine I'll have 54 metrics in each collection. I would like to group `machine_cpu_cache_capacity_bytes`, `machine_node_hugepages_count`, `machine_node_memory_capacity_bytes` and `machine_thread_siblings_count` metrics as `cpu_topology` and make them disable on default. I think about adding `cpu_topology` as an option to `--diable-metrics` parameter. What do you think?

Reply:
Thats a great idea.